.github/workflows/cli-docs-update.yml

```diff
@@ -0,0 +1,36 @@
+name: CLI commands docs check
+
+on:
+  pull_request:
+    branches:
+      - master
+      - support/**
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: cli-docs-check
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Get changed CLI-related files
+        id: cli-diff
+        uses: tj-actions/changed-files@v42
+        with:
+          files: |
+            cmd/neofs-cli/**
+
+      - name: Get changed CLI doc files
+        id: docs-diff
+        uses: tj-actions/changed-files@v42
+        with:
+          files: docs/cli-commands/**
+
+      - name: Fail if CLI files are changed but the documentation is not updated
+        if: steps.cli-diff.outputs.any_changed == 'true' && steps.docs-diff.outputs.any_changed == 'false'
```

Comment:
I can change a lot of things in `cmd/neofs-cli` that won't affect documentation in any way. Better regenerate (and maybe even have some `Makefile` target for this) and check for diffs.